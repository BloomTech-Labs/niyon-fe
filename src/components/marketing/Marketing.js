import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import MenuIcon from '@material-ui/icons/Menu';
import CloseIcon from '@material-ui/icons/Close';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper'
import './styles.scss';


const drawerWidth = 240;
const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0,
    },
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('sm')]: {
      display: 'none',
    },
  },
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  closeMenuButton: {
    marginRight: 'auto',
    marginLeft: 0,
  },
  buttons: {
    margin: "20% auto 0 auto",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },
  button: {
    width: "250px",
    borderRadius: "50px",
    height: "50px",
    margin: "20px 0"
  },
  paper: {
    backgroundImage: `url(${"static/src/assets/images/niyon_marketing_header.jpg"})`,  
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    backgroundPosition: "left top",
    width: "500px",
    height: "300px",
    margin: "20% 0"
}
}));


function Marketing() {
  const dummyCategories = ['Signup', 'Login']
  const classes = useStyles();
  const theme = useTheme();
  const [mobileOpen, setMobileOpen] = React.useState(false);
function handleDrawerToggle() {
    setMobileOpen(!mobileOpen)
  }
const drawer = (
    <div>
      <List>
        {dummyCategories.map((text, index) => (
          <ListItem button key={text}>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
    </div>
  );
return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="Open drawer"
            edge="start"
            onClick={handleDrawerToggle}
            className={classes.menuButton}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            Responsive drawer
          </Typography>
        </Toolbar>
      </AppBar>
      
      <nav className={classes.drawer}>
        {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
        <Hidden smUp implementation="css">
          <Drawer
            variant="temporary"
            anchor={theme.direction === 'rtl' ? 'right' : 'left'}
            open={mobileOpen}
            onClose={handleDrawerToggle}
            classes={{
              paper: classes.drawerPaper,
            }}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
          >
            <IconButton onClick={handleDrawerToggle} className={classes.closeMenuButton}>
              <CloseIcon/>
            </IconButton>
            {drawer}
          </Drawer>
        </Hidden>
<Hidden xsDown implementation="css">
          <Drawer
            className={classes.drawer}
            variant="permanent"
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            <div className={classes.toolbar} />
            {drawer}
          </Drawer>  
        </Hidden>
      </nav>
      <div className="marketing-container">
        <div className="container"></div>      
        <div className={classes.buttons}>
        <Button variant="contained" color="primary" className={classes.button}>
          Sign Up
        </Button>
        <Button variant="contained" color="secondary" className={classes.button}>
          Login
        </Button>
        </div>
      </div>
      

    </div>
  );
}
export default Marketing;