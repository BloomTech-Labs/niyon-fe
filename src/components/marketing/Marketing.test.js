import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import Marketing from './Marketing';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import MenuIcon from '@material-ui/icons/Menu';
import CloseIcon from '@material-ui/icons/Close';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import findByTestAttr from '../../tests/utils';

const setUp = (props={}) => {
   const wrapper = shallow(<Marketing />);
   return wrapper;
}

describe('<Marketing /> component testing', () => {
   let component;
   beforeEach(() => {
      component = setUp();
   });
   it('should pass snapshot testing correctly', () => {
             expect(toJSON(component)).toMatchSnapshot();
   });

   it('should render <Marketing /> component correctly', () => {
         const marketingWrapper = findByTestAttr(component, 'marketing-page');
         expect(marketingWrapper.length).toBe(1);
   });

   it('should render  <CssBaseline /> component with no errors', () => {
         expect(component.find(CssBaseline)).toHaveLength(1);
   });

   it('should render <AppBar /> component correctly', () => {
         expect(component.find(AppBar)).toHaveLength(1);
    });
   it('should render <Toolbar /> component correctly', () => {
          expect(component.find(Toolbar)).toHaveLength(1);
   });
   it('should render <IconButton /> component correctly', () => {
        expect(component.find(Toolbar).find(IconButton)).toHaveLength(1);
    });

   it('should render <IconButton /> component correctly', () => {
         expect(component.find(Toolbar).find(IconButton)).toHaveLength(1);
   });

   it('should render <MenuIcon /> component correctly', () => {
         expect(component.find(MenuIcon)).toHaveLength(1);
   });

   it('should render logo div correctly', () => {
        const logoWrapper = findByTestAttr(component, 'test-logo');
        expect(logoWrapper.exists()).toBe(true);
     });

   it('should render title content with no errors', () => {
         expect(component.find(Typography)).toHaveLength(1);
         expect(component.find(Typography).text()).toEqual('Niyon');
     });

     it('should render navbar component correctly',() => {
         expect(component.find('.makeStyles-drawer-2').exists()).toBe(true);
     });

     describe('First <Hidden /> component', () => {
      it('should render <Hidden /> with smUp attribute correctly', () => {
            const hiddenWrapper1 = component.find(Hidden).at(0);
            expect(hiddenWrapper1.exists()).toBe(true);
            expect(hiddenWrapper1.props().smUp).toEqual(true);           
      });
      it('should render the <Drawer /> component correctly', () => {
            const hiddenWrapper1 = component.find(Hidden).at(0);
            expect(hiddenWrapper1.find(Drawer)).toHaveLength(1);
      });
       it('should render <IconButton /> inside <Drawer /> component correctly', () => {
             const hiddenWrapper1 = component.find(Hidden).at(0);
             expect(hiddenWrapper1.find(Drawer).find(IconButton)).toHaveLength(1);
       });
     });

     describe('Second <Hidden /> component', () => {
      it('should render <Hidden /> with xsDown attribute correctly', () => {
            const hiddenWrapper2 = component.find(Hidden).at(1);
            expect(hiddenWrapper2.exists()).toBe(true);
            expect(hiddenWrapper2.props().xsDown).toEqual(true);           
      });
      it('should render the <Drawer /> component correctly', () => {
            const hiddenWrapper2 = component.find(Hidden).at(1);
            expect(hiddenWrapper2.find(Drawer)).toHaveLength(1);
      });
       it('should render a <Div /> inside <Drawer /> component correctly', () => {
             const hiddenWrapper2 = component.find(Hidden).at(1);
             console.log(hiddenWrapper2.dive().debug());
             expect(hiddenWrapper2.find('.makeStyles-toolbar-5').exists()).toBe(true);
       });
     });
    
   
})