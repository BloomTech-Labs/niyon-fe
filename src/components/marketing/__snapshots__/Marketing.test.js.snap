// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Marketing /> component testing should pass snapshot testing correctly 1`] = `
<div
  className="makeStyles-root-1"
  data-test="marketing-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-3"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <div
        className="logo"
        data-test="test-logo"
      />
      <WithStyles(ForwardRef(Typography))
        noWrap={true}
        variant="h6"
      >
        Niyon
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <nav
    className="makeStyles-drawer-2"
  >
    <Hidden
      implementation="css"
      smUp={true}
    >
      <WithStyles(ForwardRef(Drawer))
        ModalProps={
          Object {
            "keepMounted": true,
          }
        }
        anchor="left"
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-6",
          }
        }
        onClose={[Function]}
        open={false}
        variant="temporary"
      >
        <WithStyles(ForwardRef(IconButton))
          className="makeStyles-closeMenuButton-8"
          onClick={[Function]}
        >
          <CloseIcon />
        </WithStyles(ForwardRef(IconButton))>
        <div>
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Home"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Home"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="About"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="About"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
    <Hidden
      implementation="css"
      xsDown={true}
    >
      <WithStyles(ForwardRef(Drawer))
        className="makeStyles-drawer-2"
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-6",
          }
        }
        variant="permanent"
      >
        <div
          className="makeStyles-toolbar-5"
        />
        <div>
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Home"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Home"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="About"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="About"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
  </nav>
  <div
    className="marketing-container"
  >
    <div
      className="container"
    >
      <h3>
        Connect with Mentors in your area!
      </h3>
    </div>
    <div
      className="makeStyles-buttons-9"
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-10"
        color="primary"
        variant="contained"
      >
        Sign Up
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-button-10"
        color="secondary"
        variant="contained"
      >
        Login
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
</div>
`;
